data_transfer/jobs/shared.py

```diff
@@ -26,3 +40,57 @@ def batch_upload_data() -> None:
                 record.is_uploaded = True
                 update_record(record)
             dmpy.rm_local_data(zip_path)
+
+
+def prepare_data_folders(device_type: DeviceType) -> None:
+    """
+    Checks folders present in 'data/upload/ are finished
+    and moves them into the upload folder in the format:
+
+        DEVICEID-PATIENTID-STARTWEAR-ENDWEAR
+    """
+    not_uploaded = records_not_uploaded(device_type)
+
+    grouped: dict = {}
+    # transform the result to {PatientID1-DeviceID1: [{Record1}, ... {Record2}], ... }
+    [
+        grouped.setdefault(f"{r.patient_id}/{r.device_id}", []).append(r)
+        for r in not_uploaded
+    ]
+
+    # filter sets which have any record with 'is_processed' == False
+    # 'is_processed' == False catches the 'False' for any preceding task as well
+    to_upload = {k: v for k, v in grouped.items() if all([r.is_processed for r in v])}
+
+    for patient_device in to_upload.keys():
+        max_data, min_data = min_max_data_wear_times(to_upload[patient_device])
+
+        start_data = max_data.strftime("%Y%m%d")
+        end_data = min_data.strftime("%Y%m%d")
+
+        source = config.storage_vol / patient_device
+        destination = (
+            config.upload_folder
+            / device_type.name
+            / f"{patient_device.replace('-','').replace('/','-')}-{start_data}-{end_data}"
```

Comment:
For consistency, might we want to have the upload folder mirror the input folder, e.g., to be `/{patient_id}/` rather than `/{deviceType}/{patient_id}/`? If yes, then device_type.name can be changed to `patient_device.split("/")[0]`

This related specifcally to line 74: -> `/ device_type.name`

Reply:
As discussed on the call yesterday, we want to have a subfolder per device pipeline - which we can mimic in the upload folder as well to make each pipeline upload their own .zip files. 

Which mean, download folder:
/device_type/patient_id/device_id
and upload folder:
/device_type/.zipfiles

I've adjusted the code to ensure the use of subfolders per device_type